satip/eumetsat.py

```diff
@@ -577,12 +581,28 @@ def create_and_download_datatailor_data(
             datatailor = eumdac.DataTailor(eumdac.AccessToken((self.user_key, self.user_secret)))
 
             # sometimes the customisation fails first time, so we try twice
-            try:
-                customisation = datatailor.new_customisation(dataset_id, chain=chain)
-            except Exception:
-                log.debug("Did not customisation first time, so trying again after 2 seconds")
-                time.sleep(2)
-                customisation = datatailor.new_customisation(dataset_id, chain=chain)
+            # This is from Data Tailor only allowing 3 customizations at once
+            # So this should then continue until it is created successfully
+            created_customization = False
+            # 5 minute timeout here
+            start = datetime.datetime.now()
+            while not created_customization and (datetime.datetime.now() - start).seconds < 600:
+                try:
+                    num_running_customizations = 0
+                    for customisation in datatailor.customisations:
+                        if customisation.status in ['INACTIVE']: # Clear stuck ones
+                            customisation.kill()
+                            customisation.delete()
+                        if customisation.status in ['RUNNING','QUEUED', 'INACTIVE']:
```

Comment:
what happens if its no in ['RUNNING','QUEUED', 'INACTIVE']

Reply:
worth putting int a logger.warning statment?

Reply:
It doesn't count towards the 3 customizations in data tailor, so doesn't matter